torch/onnx/symbolic_opset9.py

```diff
@@ -2333,33 +2475,36 @@ def _layer_norm_returns_normalized_input_mean_rstd(
 
 @symbolic_helper.quantized_args(True, False, False, False)
 @symbolic_helper.parse_args("v", "is", "v", "v", "f")
+@_beartype.beartype
 def native_layer_norm(g, input, normalized_shape, weight, bias, eps):
     return _layer_norm_returns_normalized_input_mean_rstd(
         g, input, normalized_shape, weight, bias, eps, False, True
     )
 
 
 @symbolic_helper.quantized_args(True, False, False, False)
-@symbolic_helper.parse_args("v", "is", "v", "v", "f", "i")
+@symbolic_helper.parse_args("v", "is", "v", "v", "f", "b")
+@_beartype.beartype
 def layer_norm(g, input, normalized_shape, weight, bias, eps, cudnn_enable):
     normalized, _, _ = _layer_norm_returns_normalized_input_mean_rstd(
         g, input, normalized_shape, weight, bias, eps, cudnn_enable, False
     )
     return normalized
 
 
-@symbolic_helper.parse_args("v", "v", "v", "v", "v", "i", "f", "f", "i")
+@symbolic_helper.parse_args("v", "v", "v", "v", "v", "b", "f", "f", "b")
+@_beartype.beartype
 def instance_norm(
     g,
     input,
     weight,
     bias,
     running_mean,
     running_var,
-    use_input_stats: int,
-    momentum: float,
-    eps: float,
-    cudnn_enabled: int,
+    use_input_stats: bool,
+    momentum: Number,
```

Comment:
We have seen cases where momentum is passed in as an int. The torch `Number` type consists of int, float and bool (which are 0 1 ints). So I thought to leverage it instead of creating our own number type or writing Union[int, float]

Reply:
I see, although I think it should got cast to float when annotated with "f" in `parse_args` though?

Reply:
It's possible then it was called by other symbolics and the arg was int. In that case (point) It'd be good to enforce float to ensure consistency (counter point) existing users may call with int and get an error

Reply:
Hmm looks like the extra layer of decorator messes up the argument types here. This is interesting. It's an even broader problem for those "parsed" arguments, since at caller they are all `Value`. I guess we could type annotate this guy here https://github.com/pytorch/pytorch/blob/a423c966a780a1fdac6a29c6d2be2a0709de2cd5/torch/onnx/symbolic_helper.py#L283 Still feels a bit sketchy.

Reply:
I am going to fix the caller (Quantized instance norm). But I think it makes sense to accept a (real) Number type for anything that is supposed to be floats. With that we reduce the risk of throwing errors because of callers we don't control. I can't think of anything harmful. WDYT?

Reply:
Even `True / 2 = 0.5`. I think we are safe

Reply:
That's ok. I'm a bit more concerned about the `parse_args` decorator altering argument type without being (or is it possible to) properly annotated.

Reply:
Resolved offline